test/integration/scheduler/local-pv-neg-affinity_test.go

```diff
@@ -0,0 +1,313 @@
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package scheduler
+
+// This file tests the VolumeScheduling feature.
+
+import (
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"net/http/httptest"
+	"testing"
+	"time"
+
+	"k8s.io/api/core/v1"
+	storagev1 "k8s.io/api/storage/v1"
+	"k8s.io/apimachinery/pkg/api/resource"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	utilfeature "k8s.io/apiserver/pkg/util/feature"
+	"k8s.io/client-go/informers"
+	clientset "k8s.io/client-go/kubernetes"
+	restclient "k8s.io/client-go/rest"
+	"k8s.io/client-go/tools/record"
+	"k8s.io/kubernetes/pkg/api/testapi"
+	"k8s.io/kubernetes/pkg/apis/core/v1/helper"
+	"k8s.io/kubernetes/pkg/controller/volume/persistentvolume"
+	"k8s.io/kubernetes/plugin/pkg/scheduler"
+	"k8s.io/kubernetes/plugin/pkg/scheduler/factory"
+	"k8s.io/kubernetes/test/integration/framework"
+)
+
+const (
+	affinityLabelKey = "kubernetes.io/hostname"
+)
+
+func TestLocalPVNegativeAffinity(t *testing.T) {
+	config := setupNodes(t, "volume-scheduling", 3)
+	defer config.teardown()
+
+	pv := makeHostBoundPV(t, "local-pv", classImmediate, "", "", "node-1")
+	pvc := makePVC("local-pvc", config.ns, &classImmediate, "")
+
+	// Create PV
+	if _, err := config.client.CoreV1().PersistentVolumes().Create(pv); err != nil {
+		t.Fatalf("Failed to create PersistentVolume %q: %v", pv.Name, err)
+	}
+
+	// Create PVC
+	if _, err := config.client.CoreV1().PersistentVolumeClaims(config.ns).Create(pvc); err != nil {
+		t.Fatalf("Failed to create PersistentVolumeClaim %q: %v", pvc.Name, err)
+	}
+
+	nodeMarkers := []interface{}{
+		markNodeAffinity,
+		markNodeSelector,
+		markNodeName,
+	}
+
+	for i := 0; i < len(nodeMarkers); i++ {
+		pod := makePod("local-pod", config.ns, []string{"local-pvc"})
+		nodeMarkers[i].(func(*v1.Pod, string))(pod, "node-2")
+		// Create Pod
+		if _, err := config.client.CoreV1().Pods(config.ns).Create(pod); err != nil {
+			t.Fatalf("Failed to create Pod %q: %v", pod.Name, err)
+		}
+
+		if err := waitForPodToSchedule(config.client, pod); err != nil {
+			// t.Errorf("Failed to schedule Pod %q: %v", pod.Name, err)
+			t.Logf("Failed to schedule Pod %q: %v", pod.Name, err)
+		}
+
+		resultingPod, err := config.client.CoreV1().Pods(config.ns).Get("local-pod", metav1.GetOptions{})
+		if err != nil {
+			t.Errorf("Failed to list pods with err: %v", err)
+		}
+		fmt.Printf("%s\n", printIndentedJson(resultingPod))
+
+		if err := config.client.CoreV1().Pods(config.ns).Delete("local-pod", &metav1.DeleteOptions{}); err != nil {
+			t.Fatalf("Failed to create Pod %q: %v", pod.Name, err)
+		}
+	}
+}
+
+func setupNodes(t *testing.T, nsName string, numberOfNodes int) *testConfig {
```

Comment:
Are you able to reuse the setup function in the volume binding test?

Reply:
If you agree, I will modify setup functions to be able to start mulitple nodes. In this case in your tests I will call it with number of nodes == 1

Reply:
done

Reply:
When I use new seupNodes in yout volume binding test, both tests fail, when I use in my PR new setupNodes and in yours setup, both tests pass. Mystery...

Reply:
That's odd. Are the node names, node labels, etc exactly the same between the two?